prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -336,7 +336,7 @@ void rxr_pkt_req_data_from_tx(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry
 	 *    1st iov for header.
 	 */
 	if ((!pkt_entry->mr || tx_entry->desc[tx_iov_index]) &&
-	    (tx_iov_offset + data_size < tx_entry->iov[tx_iov_index].iov_len)) {
+	    (tx_iov_offset + data_size <= tx_entry->iov[tx_iov_index].iov_len)) {
 
 		assert(ep->core_iov_limit >= 2);
 		pkt_entry->send->iov[0].iov_base = pkt_entry->pkt;
```

Comment:
This is for line 344 (sorry github isn't letting me comment on that line)

Is the conditional on `pkt_entry->mr` still required? Shouldn't that always be registered and set? Seems like that's old utility provider code bits.

Reply:
pkt_entry->mr is not set if shm provider is used